### ISSUE TITLE
Unique Job Dependencies Injection Bug

### ISSUE THREAD

**Issue by exodusanto** [lorisleiva/laravel-actions#95]
Hi, I found a new bug for AsJob case,

Using `ShouldBeUnique` contract the `JobDecorator $job` dependency injection behavior for `asJob` method doesn't work.

### Example:

`SomeWebhook.php`
```php
...
CreateVideo::dispatch($video->id)
...
```

`CreateVideo.php`
```php
...
class CreateVideo implements ShouldQueue, ShouldBeUnique
{
    use AsAction;

    public int $jobUniqueFor = 3600;

    public function getJobUniqueId($video)
    {
        return $video;
    }

    public function configureJob(JobDecorator $job): void
    {
        $job->setTries(10);
    }
...

    public function asJob(JobDecorator $job, $video): void
    {
        try {
            $this->handle(Video::find($video));
        } catch (SomeCustomException $e) {
            $job->release(60);
        }
    }
}
```

### Exception:
```
asJob() must be an instance of Lorisleiva\Actions\Decorators\JobDecorator, int given
```

**Comment by leandrodiogenes** [comment]
try this

```php

    public function asJob(\Lorisleiva\Actions\Decorators\UniqueJobDecorator $job, $video): void
    {
        try {
            $this->handle(Video::find($video));
        } catch (SomeCustomException $e) {
            $job->release(60);
        }
    }
```

**Comment by exodusanto** [comment]
@leandrodiogenes yes! 

Thank you, 
I supposed that  I can use `JobDecorator` cause `UniqueJobDecorator` extend it, but it seems not.

**Comment by lorisleiva** [comment]
Hi there 👋 

I've fixed this and released it as [`v2.1.2`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.1.2). 🚀 

You should now be able to typehint the job using `UniqueJobDecorator`.

Let me know if that all works for you. Thanks for raising this! 🍺
